app/Services/ReservationReassignmentService.php

```
@@ -0,0 +1,581 @@
+        }
+
+        $sql .= " LIMIT 1";
+
+        $stmt = $this->db->prepare($sql);
+        $stmt->bind_param($types, ...$params);
+        $stmt->execute();
+        $res = $stmt->get_result()->fetch_assoc();
+        $stmt->close();
+
+        return $res ? (int) $res['id'] : null;
+    }
```

PHPMD (2.15.0)

Avoid unused private methods such as 'findAvailableCopy'. (undefined)

(UnusedPrivateMethod)